src/test/scala/mesosphere/marathon/api/v2/AppsResourceTest.scala

```diff
@@ -218,40 +219,256 @@ class AppsResourceTest extends MarathonSpec with MarathonActorSupport with Match
+    )
+
+    Then("The return code indicates that the hostPath of volumes[0] is missing") // although the wrong field should fail
+    response.getStatus should be(422)
+    response.getEntity.toString should include("/container/volumes(0)/external/provider")
+    response.getEntity.toString should include("is unknown provider")
+  }
+
+  test("Creating an app with an external volume with no name provider name specified should FAIL provider validation") {
+    Given("An app with an unnamed volume provider")
+    val e = intercept[JsResultException]{
+      createAppWithVolumes("MESOS",
+        """
+          |    "volumes": [{
+          |      "containerPath": "/var",
+          |      "external": {
+          |        "size": 10
+          |      },
+          |      "mode": "RW"
+          |    }]
+        """.stripMargin
+      )
+    }
+
+    Then("The it should fail with a JsResultException")
+    e.getMessage should include("/container/volumes(0)/external/provider")
+    e.getMessage should include("/container/volumes(0)/external/name")
+  }
+
+  test("Creating an app with an external volume and MESOS containerizer should pass validation") {
+    Given("An app with a named, non-'agent' volume provider")
+    val response = createAppWithVolumes("MESOS",
+      """
+        |    "volumes": [{
+        |      "containerPath": "/var",
+        |      "external": {
+        |        "size": 10,
+        |        "provider": "dvdi",
+        |        "name": "namedfoo",
+        |        "options": {"dvdi/driver": "bar"}
+        |      },
+        |      "mode": "RW"
+        |    }]
+      """.stripMargin
+    )
+
+    Then("The return code indicates create success")
+    response.getStatus should be(201)
+  }
+
+  test("Creating an app with an external volume using an invalid rexray option should fail") {
+    Given("An app with a named, non-'agent' volume provider")
+    val response = createAppWithVolumes("MESOS",
+      """
+        |    "volumes": [{
+        |      "containerPath": "/var",
+        |      "external": {
+        |        "size": 10,
+        |        "provider": "dvdi",
+        |        "name": "namedfoo",
+        |        "options": {"dvdi/driver": "rexray", "dvdi/iops": "0"}
+        |      },
+        |      "mode": "RW"
+        |    }]
+      """.stripMargin
+    )
+
+    Then("The return code indicates validation error")
+    response.getStatus should be(422)
+    response.getEntity.toString should include("/container/volumes(0)/external.options/dvdi/iops")
+  }
+
+  test("Creating an app with an external volume and DOCKER containerizer should pass validation") {
+    Given("An app with a named, non-'agent' volume provider")
+    val response = createAppWithVolumes("DOCKER",
+      """
+        |    "volumes": [{
+        |      "containerPath": "/var",
+        |      "external": {
+        |        "provider": "dvdi",
+        |        "name": "namedfoo",
+        |        "options": {"dvdi/driver": "bar"}
```

Comment:
I question whether this should pass validation because the options are currently ignored. @jdef @sttts